api_fhir_r4/converters/activityDefinitionConverter.py

```diff
@@ -119,301 +156,247 @@ def build_fhir_title(cls, fhir_activity_definition, imis_activity_definition):
-            codeable_concept.text = "Adult or Kid"
+        codeable_concept.coding.append(cls.build_fhir_workflow_coding(imis_activity_definition.category))
+        codeable_concept.text = codeable_concept.coding[0].display
 
         return codeable_concept
 
     @classmethod
     def build_fhir_venue(cls, imis_activity_definition):
-        display = ""
-        if imis_activity_definition.care_type == "O":
-            display = "Out-patient"
-        if imis_activity_definition.care_type == "I":
-            display = "In-patient"
-        if imis_activity_definition.care_type == "B":
-            display = "Both"
-
         codeable_concept = CodeableConcept.construct()
-        coding_venue = Coding.construct()
-        coding_venue.code = imis_activity_definition.care_type
-        coding_venue.display = display
-        if type(codeable_concept.coding) is not list:
-            codeable_concept.coding = [coding_venue]
-        else:
-            codeable_concept.coding.append(coding_venue)
-        codeable_concept.text = "Clinical Venue"
+        codeable_concept.coding = []
+
+        ambulatory_venue = imis_activity_definition.care_type in [Service.CARE_TYPE_OUT_PATIENT, Service.CARE_TYPE_BOTH]
+        if ambulatory_venue:
+            ambulatory_coding = cls.build_fhir_venue_coding(Service.CARE_TYPE_OUT_PATIENT)
+            codeable_concept.coding.append(ambulatory_coding)
+
+        imp_venue = imis_activity_definition.care_type in [Service.CARE_TYPE_IN_PATIENT, Service.CARE_TYPE_BOTH]
+        if imp_venue:
+            imp_coding = cls.build_fhir_venue_coding(Service.CARE_TYPE_IN_PATIENT)
+            codeable_concept.coding.append(imp_coding)
+
+        codeable_concept.text = " or ".join([coding.display for coding in codeable_concept.coding])
+
         return codeable_concept
 
     @classmethod
-    def build_fhir_level(self, imis_activity_definition: Service):
-        # Values for this extension are fixed for medication
-        display = ""
-        if imis_activity_definition.level == 'S':
-            display = 'Simple Service'
-        elif imis_activity_definition.level == 'V':
-            display = 'Visit'
-        elif imis_activity_definition.level == 'D':
-            display = 'Day of stay'
-        elif imis_activity_definition.level == 'H':
-            display = 'Hospital case'
-
+    def build_fhir_gender(cls, imis_activity_definition):
         codeable_concept = CodeableConcept.construct()
-        coding_level = Coding.construct()
-        coding_level.code = imis_activity_definition.level
-        coding_level.display = display
-        if type(codeable_concept.coding) is not list:
-            codeable_concept.coding = [coding_level]
-        else:
-            codeable_concept.coding.append(coding_level)
-        codeable_concept.text = "Service Level"
+        codeable_concept.coding = []
+
+        male_flag = PatientCategoryMapping.imis_patient_category_flags["male"]
```

Comment:
@dragos-dobre  I think we should no put any usecontext is it is open to all and have a specific "blocked" context if none is activated

Reply:
Note taken. The "blocked" context should be validated when adding Items and Services in openIMIS.